src/aura/autocomplete/autocompleteHelper.js

```diff
@@ -71,5 +76,106 @@
+    handleArrowDownKey: function (component, helper) {
+        const listbox = document.querySelector('[role="listbox"]');
+        const elements = component.get("v.items");
+        var currentFocussedElement = listbox.getAttribute('aria-activedescendant');
+        
+        if (currentFocussedElement == null || currentFocussedElement == '') {
+            if (elements[0]) {
+                currentFocussedElement = elements[0].value.Id
+                listbox.setAttribute('aria-activedescendant', currentFocussedElement);
+                helper.fireNewItemFocus(currentFocussedElement);
+                return;
+            } else {
+                var showFooter = component.get('v.showListFooter');
+                if (showFooter) {
+                    helper.fireFocusOnFooter(component, listbox);
+                    return;
+                }
+            }
+            
+        }
+
+        var newFocussedElement = null; 
+        
+        for (var i = 0; i < elements.length; i++) {
+            if (elements[i].value.Id == currentFocussedElement) {
+                // I am in the middle of the list, setting the next descendant to the next element
+                if (i < elements.length - 1) {
+                    newFocussedElement = elements[i + 1].value.Id;
+                    listbox.setAttribute('aria-activedescendant', newFocussedElement);
+                    break;
+                } else {
+                    // reach the last element of the list, next one is footer(if present) or the first element if not present.
+                    var showFooter = component.get('v.showListFooter');
+                    if (showFooter) {
+                        helper.fireFocusOnFooter(component, listbox);
+                        return;
+                    } else {
+                        newFocussedElement =  elements[0].value.Id;
+                        listbox.setAttribute('aria-activedescendant', newFocussedElement);
+                        break;
+                    }
+                }
+            }
+        }
+        
+        helper.fireNewItemFocus(newFocussedElement);
+    },
+    // Handles the pressing of the arrow up key.
+    // It updates the aria-activedescendant and notify the child elements
+    handleArrowUpKey: function (component, helper) {
+        const listbox = document.querySelector('[role="listbox"]');
+        const elements = component.get("v.items");
+        var currentFocussedElement = listbox.getAttribute('aria-activedescendant');
+        
+        if (currentFocussedElement == null || currentFocussedElement == '') {
+            if (elements[elements.length - 1]) {
+                currentFocussedElement = elements[elements.length - 1].value.Id
+                listbox.setAttribute('aria-activedescendant', currentFocussedElement);
+                helper.fireNewItemFocus(currentFocussedElement);
+                return;
+            }
+        }
+        var newFocussedElement = null; 
```

Comment:
This is not used until the for loop below.  Can we move this to immediately before that for loop and add a blank line here?